### PR TITLE
converted dns_lookup_adapter_resolver_pool_size warning message to use dynamic value instead of hard-coded value (10)

### PR DESCRIPTION
**Pull request by williamtrelawny**
## Description
The value of dns_lookup_adapter_resolver_pool_size in the warning message for when this value should be increased should be dynamic and based on the actual value configured in server.conf. Previously, it was hard-coded to the default of "10".

## Motivation and Context
Fixes inaccurate feedback to user who has modified the dns_lookup_adapter_resolver_pool_size value. Now the user will at least receive acknowledgement that their modified value is in use by Graylog.

## How Has This Been Tested?
Not tested unfortunately as I do not have the knowledge/ability to. However this being a very minor change, I felt it appropriate to submit this PR anyway and welcome someone to test it who has the ability to.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by ryan-carroll-graylog — CHANGES_REQUESTED** [review]
@williamtrelawny Looks great, thanks for knocking this out!

I think this does warrant a changelog entry since it is a user facing change in behavior. There's more info about changelogs here: https://graylogdocumentation.atlassian.net/wiki/spaces/GLE/pages/2518908931/Changelog+Entries

The TLDR is that you should add a changelog file in `graylog2-server/changelog/unreleased/` to this PR.

Other than that this looks good and tests successfully! I moved the logging slightly to test and the results look good to me: 
```
2023-09-14 11:43:34,334 WARN : org.graylog2.lookup.adapters.dnslookup.DnsResolverPool - Lease for resolver [04a8099f-1283-4627-b187-e8bea687b907] is in-use. Skipping refresh. This will be attempted again in [10] seconds. If this happens frequently for high message rates, consider increasing the [dns_lookup_adapter_resolver_pool_size = 99] server configuration property to allow more DNS resolvers.
```

**Review by danotorrey — APPROVED** [review]
LGTM and tests successfully! Well done @williamtrelawny! Good call on the change log @ryan-carroll-graylog!

**Comment by williamtrelawny** [comment]
Sorry about the force-push - I hit the darn "sync forks" button in the browser and muddled the commit history of this PR with an unrelated commit. 

Fixed now, adding the changelog entry now! Thanks for your all's support 😄

**Review by ryan-carroll-graylog — APPROVED** [review]
Thanks for the update @williamtrelawny!

File: changelog/unreleased/pr-16475.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+type = "f"
+message = "server.log now accurately reflects configured `dns_lookup_adapter_resolver_pool_size` value."
```
Comment (danotorrey) [inline changelog/unreleased/pr-16475.toml:2]: Nitpick: Suggesting a slightly more specific wording.
```suggestion
message = "DNS Resolver Pool warning log message now accurately reflects configured `dns_lookup_adapter_resolver_pool_size` value."
```

**Comment by williamtrelawny** [comment]
Sorry again- I had the wrong email address in the signed-off-by so the CLA check failed... Amended and should all be good now!

Next PR won't be this messy, I swear! 😅

**Comment by danotorrey** [comment]
All good @williamtrelawny! Fork PRs are always a bit challenging. Nice work!

**Review by danotorrey — APPROVED** [review]

**Review by kingzacko1 — APPROVED** [review]

**Pull request merged**